### PR TITLE
Add support for shadow-cljs

### PR DESCRIPTION
**Pull request by zeitstein**

### CODE REVIEW

File: src/repley/shadow/preload.cljs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+(ns repley.shadow.preload
+  "May be added to `shadow-cljs.edn` `:preloads` to add tap on start."
+  (:require
+   [repley.shadow.remote :as r]
+   goog.object))
+
+(let [port (-> (goog.object/get js/CLOSURE_DEFINES "repley.shadow.build.opts")
+               (js->clj :keywordize-keys true)
+               (get-in [:http :port]))
```
Comment (zeitstein) [inline src/repley/shadow/preload.cljs:9]: `fetch` should use the port defined in `shadow-cljs.edn`.

File: src/repley/shadow/preload.cljs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      opts (cond-> {:mode "no-cors"}
```
Comment (zeitstein) [inline src/repley/shadow/preload.cljs:10]: I've had to add `no-cors` to make it work on my machine, but perhaps this shouldn't be the default?

At any rate, `repley.shadow.preload` and `repley.shadow.build` may be considered as a blueprint for users to customize further. (Example: I'm using a modified `repley.shadow.build/start-and-listen` in order to add readers for edn.)

File: src/repley/shadow/remote.cljs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+(ns repley.shadow.remote)
+
+(defn make-send [fetch]
+  (fn submit
+    ([value] (submit nil value))
+    ([{:keys [port mode]
+       :or   {port 3001 ;; TODO: use repley.config/default-config
```
Comment (zeitstein) [inline src/repley/shadow/remote.cljs]: `repley.config` would need to be `.cljc`.

Comment (tatut) [reply src/repley/shadow/remote.cljs]: Config is only data and a simple merging function, changed that to .cljc

File: src/repley/shadow/remote.cljs [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 
 (defn make-send [fetch]
   (fn submit
     ([value] (submit nil value))
     ([{:keys [port mode]
-       :or   {port 3001 ;; TODO: use repley.config/default-config
+       :or   {port default-port
               mode "cors"}}
       value]
      (fetch (str "http://localhost:" port "/receive")
```
Comment (tatut) [inline src/repley/shadow/remote.cljs:14]: should probably get `:prefix` and `:receive-endpoint` as they constitute parts of the URL as well

Comment (zeitstein) [reply src/repley/shadow/remote.cljs:14]: Which part of the URL is the prefix exactly?

Btw, the `merge-with merge` strategy is not going to work for strings like `:prefix`.

Comment (tatut) [reply src/repley/shadow/remote.cljs:14]: oh yeah, good catch... :) 

the prefix is meant for allowing you to use some other path for the URLs, so it is the beginning.

**Pull request merged**